File: src/creational/abstract_factory/bombed/BombedMazeFactory.java
package creational.abstract_factory.bombed;

import creational.abstract_factory.base.MazeFactory;
import creational.abstract_factory.base.Room;
import creational.abstract_factory.base.Wall;

public class BombedMazeFactory extends MazeFactory {
    @Override
    public Wall makeWall() {
        return new BombedWall();
    }

    @Override
    public Room makeRoom(int n) {
        return new RoomWithABomb(n);
    }

}

File: src/creational/abstract_factory_kotlin/bombed/BombedMazeFactory.kt
package creational.abstract_factory_kotlin.bombed

import creational.abstract_factory_kotlin.MazeFactory

class BombedMazeFactory : MazeFactory() {
    override fun makeWall() = BombedWall()
    override fun makeRoom(n: Int) = RoomWithABomb(n)
}
